### PR TITLE
COMPAT: Update thrift

### LINKED ISSUE #261: Fastparquet fails on reading pyarrow written metadata on Windows
I am enabling the cross-compatibility test in pandas in https://github.com/pandas-dev/pandas/pull/18662, and the test for reading with `fastparquet` of file written by `pyarrow` is failing on Windows. 

The frame is

```
df = pd.DataFrame({'a': list('abc'),
                       'b': list(range(1, 4)),
                       'd': np.arange(4.0, 7.0, dtype='float64'),
                       'e': [True, False, True],
                       'f': pd.date_range('20130101', periods=3),
                      })

df.to_parquet(path, engine=pa, compression=None)
result = read_parquet(path, engine=fp)
```

The error is (from https://ci.appveyor.com/project/pandas-dev/pandas/build/1.0.7727/job/y2xr7bydcttm9p80):
```
================================== FAILURES ===================================
___________________________ test_cross_engine_pa_fp ___________________________
[gw0] win32 -- Python 3.6.3 C:\Miniconda3_64\envs\pandas\python.exe
df_cross_compat =    a  b    d      e          f
0  a  1  4.0   True 2013-01-01
1  b  2  5.0  False 2013-01-02
2  c  3  6.0   True 2013-01-03
pa = 'pyarrow', fp = 'fastparquet'
    def test_cross_engine_pa_fp(df_cross_compat, pa, fp):
        # cross-compat with differing reading/writing engines
    
        df = df_cross_compat
        with tm.ensure_clean() as path:
            df.to_parquet(path, engine=pa, compression=None)
    
>           result = read_parquet(path, engine=fp)
pandas\tests\io\test_parquet.py:184: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
pandas\io\parquet.py:211: in read_parquet
    return impl.read(path, columns=columns, **kwargs)
pandas\io\parquet.py:123: in read
    return self.api.ParquetFile(path).to_pandas(columns=columns, **kwargs)
C:\Miniconda3_64\envs\pandas\lib\site-packages\fastparquet\api.py:102: in __init__
    self._parse_header(f, verify)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
self = <[AttributeError("'ParquetFile' object has no attribute '_schema'") raised in repr()] ParquetFile object at 0xa5e425bc50>
f = <_io.BufferedReader name='C:\\Users\\appveyor\\AppData\\Local\\Temp\\1\\tmpltg33jat'>
verify = False
    def _parse_header(self, f, verify=True):
        try:
            f.seek(0)
            if verify:
                assert f.read(4) == b'PAR1'
            f.seek(-8, 2)
            head_size = struct.unpack('<i', f.read(4))[0]
            if verify:
                assert f.read() == b'PAR1'
        except (AssertionError, struct.error):
            raise ParquetException('File parse failed: %s' % self.fn)
    
        f.seek(-(head_size+8), 2)
        try:
            fmd = read_thrift(f, parquet_thrift.FileMetaData)
        except Exception:
            raise ParquetException('Metadata parse failed: %s' %
>                                  self.fn)
E           fastparquet.util.ParquetException: Metadata parse failed: C:\Users\appveyor\AppData\Local\Temp\1\tmpltg33jat
```

Again, I don't know whether this is a fastparquet or pyarrow bug (but reporting here is easier than on jira, sorry for that)

### PR DESCRIPTION
**Pull request by TomAugspurger**
Based off

https://github.com/apache/arrow/blob/97678c106054c4d9a738ddcb1c895a6ab7ca9d31/python/pyarrow/_parquet.pxd

Closes #261

### CODE REVIEW

File: fastparquet/parquet.thrift [hunk 1]
```diff
@@ -323,6 +363,8 @@ enum CompressionCodec {
   GZIP = 2;
   LZO = 3;
   BROTLI = 4;
+  LZ4 = 5;
```
Comment (martindurant) [inline fastparquet/parquet.thrift]: Note for the future: these compressions are not implemented (but are trivial)

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,2 @@
+fastparquet/parquet_thrift: fastparquet/parquet.thrift
```
Comment (TomAugspurger) [inline Makefile]: @martindurant not sure what your thoughts are on adding a Makefile. I can never remember where the dashes go the `thrift` commands though :)

Comment (martindurant) [reply Makefile]: Does this squash the existing generated files each time?
I am no expert on `make`; but will this work on every platform?

Comment (TomAugspurger) [reply Makefile]: > Does this squash the existing generated files each time?

You can run `make` whenever, it'll only do anything when `parquet.thrift` changes. It's identical to just running those commands manually.

>  I am no expert on make; but will this work on every platform?

Linux / mac yes (maybe Mac needs `xcode-select` installed, can't remember).

Windows, yes with some work getting the PATH configured correctly. Even if it doesn't work for contributors, I think having it here is useful just to remember the command.

Comment (martindurant) [reply Makefile]: OK, then no objection from me.

**Comment by martindurant** [comment]
I have no further comments, happy for this to merge.

**Pull request merged**